### PR TITLE
First pass at backend docs for #228

### PR DESCRIPTION
**Pull request by Drakulix**
Probably contains a few spelling mistakes, but I think I caught everything that was out-dated at least.
I will give this another go tomorrow. :)

### CODE REVIEW

File: src/backend/egl/mod.rs [hunk 1]
```diff
@@ -1,30 +1,30 @@
 //! Common traits and types for egl rendering
 //!
-//! Large parts of this module are taken from
-//! [glutin src/api/egl](https://github.com/tomaka/glutin/tree/044e651edf67a2029eecc650dd42546af1501414/src/api/egl/)
+//! This module has multiple responsibilities related to functionality provided by libEGL:
+//! - Initializing EGL objects to
+//!   - initialize usage of EGL based [`WlBuffer`](wayland_server::protocol::wl_buffer::WlBuffer)s via `wl_drm`.
+//!   - initialize OpenGL contexts from.
+//!   - Import/Export external resources to/from OpenGL
```
Comment (vberger) [inline src/backend/egl/mod.rs:7]: It seems there is one too many layers of lists here? The topmost list has only one item.

**Review by vberger — APPROVED** [review]
Just a tiny nitpick, but otherwise LGTM. :+1: 

Which modules does that cover, so that we can keep track of where we are at on #228

**Comment by Drakulix** [comment]
> Just a tiny nitpick, but otherwise LGTM. +1
> 
> Which modules does that cover, so that we can keep track of where we are at on #228

Should be everything that is missing, although I found most out of date docs in the egl-module.
But I was a little tired yesterday, so I will look through all of it a second time.

**Comment by Drakulix** [comment]
Alright I am done with this one. Fixes #228

**Review by vberger — APPROVED** [review]

**Pull request merged**